src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java

```diff
@@ -116,13 +118,18 @@ public BitbucketSCMSource(
                                                 repositoryName,
                                                 mirrorName));
                 setRepositoryDetails(credentialsId, sshCredentialsId, serverId, mirroredRepository);
+                selfLink = mirroredRepository.getRepository().getSelfLink();
             } catch (MirrorFetchException ex) {
                 setEmptyRepository(credentialsId, sshCredentialsId, projectName, repositoryName, serverId, mirrorName);
             }
         } else {
             BitbucketRepository localRepo = scmHelper.getRepository(projectName, repositoryName);
             setRepositoryDetails(credentialsId, sshCredentialsId, serverId, "", localRepo);
+            selfLink = localRepo.getSelfLink();
         }
+        //self link contains `/browse` which we must trim off.
+        String repositoryUrl = selfLink.substring(0, max(selfLink.indexOf("/browse"), 0));
```

Comment:
/supernit I think we want `selfLink.lastIndexOf("/browse")`, in the case the repo or project name is also browse

Reply:
I copied this from the `BitbucketSCM` happy to change there as well if you think it worth it?

Reply:
/shrug yeah given it's such a simple fix

Reply:
Yep, I've pushed the change, once builds are green I'll merge